### PR TITLE
General GraphQL resolver optimizations

### PR DESCRIPTION
**Pull request by noxer**
#### Summary
This PR contains a bunch of small optimizations, both performance- and code-wise, in the GraphQL resolvers. Nothing fancy, just a bunch of low hanging fruit.

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by agnivade — APPROVED** [review]
Great work :)

File: api4/resolver_channel_test.go [hunk 1]
```diff
@@ -436,7 +436,7 @@ func TestGetPrettyDNForUsers(t *testing.T) {
 				LastName:  "last2",
 			},
 		}
-		assert.Equal(t, "nick2", getPrettyDNForUsers("nickname_full_name", users, "user1"))
+		assert.Equal(t, "nick2", getPrettyDNForUsers("nickname_full_name", users, "user1", map[string]string{}))
```
Comment (larkox) [inline api4/resolver_channel_test.go:439]: Do we need / want tests to check the cache is working as expected?

**Review by larkox — APPROVED** [review]
LGTM, just one question regarding tests.

File: api4/resolver_channel_test.go (hunk 1 above)
Comment (agnivade) [reply api4/resolver_channel_test.go:439]: Don't think that's necessary.

Comment (noxer) [reply api4/resolver_channel_test.go:439]: Just to be sure this will work in the future, I'll add a small check.

**Pull request merged**